### DISCUSSION TITLE
eslint with typescript lints successfully but fails to display outputs.

### DISCUSSION THREAD

**Discussion by hardikkheni** [eslint/eslint#17163] [Q&A] [answered]
There was a problem loading formatter: /Users/hardikkheni/Workspace/personal/node-boilderplates/node_modules/eslint/lib/cli-engine/formatters/stylish
Error: require() of ES Module /Users/hardikkheni/Workspace/personal/node-boilderplates/node_modules/strip-ansi/index.js from /Users/hardikkheni/Workspace/personal/node-boilderplates/node_modules/eslint/lib/cli-engine/formatters/stylish.js not supported.
Instead change the require of index.js in /Users/hardikkheni/Workspace/personal/node-boilderplates/node_modules/eslint/lib/cli-engine/formatters/stylish.js to a dynamic import() which is available in all CommonJS modules.
<img width="2052" alt="Screenshot 2023-05-08 at 12 44 47 PM" src="https://user-images.githubusercontent.com/19410263/236776311-bd9d6f2b-1e05-4757-a093-eca42d222e67.png">

**Comment by Rec0iL99** [comment]
Hi @hardikkheni, can you please provide a small reproducible example for us to try out and see where the problem is? I tried reproducing locally and found no problems. Maybe you were trying something else. Thanks.

**Reply by hardikkheni** [reply]
i failed to replicate in stackblitz, but when i run lint command with terminal it works but if i run from npm script it fails with error.

**Reply by hardikkheni** [reply]
when i run ```jest --coverage``` it fails with same error from eslint in eslint package.

it suggests that i change dynamic import from es module require.

**Reply by Rec0iL99** [reply]
@hardikkheni, can you please put the example in a Github repo and post the link for us to try out? Please also provide the steps needed to reproduce the error. Thanks.

**Reply by hardikkheni** [reply]
here is the repo to repoduce the issue.
https://github.com/hardikkheni/node-boilderplates

**Reply by hardikkheni** [reply]
which preset i should use?
1). https://github.com/standard/eslint-config-standard-with-typescript (which suggested by eslint prompt)
2). airbnb

i have read somewhere that airbnb is not being maintain that why eslint removed from their suggestion lint in prompt, is that true?

**Reply by Rec0iL99** [reply]
Hi @hardikkheni, thanks for providing the example for me to test.

I couldn't reproduce the issue you were facing.

> but when i run lint command with terminal it works but if i run from npm script it fails with error

Here are my results and ESLint works fine for me:

```bash
# I added the @typescript-eslint/no-unused-vars rule with error severity to check if ESLint reported any problems with rules enabled
> npx eslint ./src --ext .ts

/Users/joelmathew/Desktop/node-boilderplates/src/app.ts
  1:7  error  'foo' is assigned a value but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

``` bash
> npm run lint

> node-boilderplates@1.0.0 lint
> eslint ./src --ext .ts


/Users/joelmathew/Desktop/node-boilderplates/src/app.ts
  1:7  error  'foo' is assigned a value but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

> when i run jest --coverage it fails with same error from eslint in eslint package.

``` bash
> npx jest --coverage
 PASS  test/index.test.ts
  App test
    ✓ object should have function called listen. (2 ms)

----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |      80 |      100 |      50 |      80 |                   
 app.ts   |      80 |      100 |      50 |      80 | 8                 
----------|---------|----------|---------|---------|-------------------
Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        1.699 s, estimated 2 s
Ran all test suites.
```

``` bash
> npm run test

> node-boilderplates@1.0.0 test
> jest --coverage

 PASS  test/index.test.ts
  App test
    ✓ object should have function called listen. (2 ms)

----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |      80 |      100 |      50 |      80 |                   
 app.ts   |      80 |      100 |      50 |      80 | 8                 
----------|---------|----------|---------|---------|-------------------
Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        1.675 s, estimated 2 s
Ran all test suites.
```

ESLint is working as expected. It looks like a problem with your local setup. I suggest you delete your `node_modules` and do a fresh install of the deps, it might help.

You are free to use any style guide. ESLint does not [show](https://github.com/eslint/create-config/pull/33) Airbnb and Google style guides for TS in prompt because TS is not officially supported by these style guides. You will find them in options if your project does not use TypeScript.

**Reply by hardikkheni** [reply]
ohhh..., i used yarn that why i am having issue. but then i switch to npm it's working....
with yarn i also try to clean install of deps but still facing the issue. may be you can replicate the issue with yarn.

**Reply by hardikkheni** [reply]
so, i have also tried to clear install with yarn, but it didn't work then switch npm and then back to yarn with clear install it worked.

but what could be the reason behind this because i have tried clear installing yarn and even with multiple setup  from scratch and still had issue.

**Reply by Rec0iL99** [reply, marked as answer]
I tried with `yarn` and faced no issues. Could not replicate the issue you are facing. 

```bash
> yarn lint

/Users/joelmathew/Desktop/node-boilderplates/src/index.ts
  1:15  error  'sub' is defined but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

```bash
> yarn test
 PASS  test/index.test.ts
  App test
    ✓ object should have function called listen. (2 ms)

----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |      75 |      100 |      50 |      75 |                   
 app.ts   |      75 |      100 |      50 |      75 | 6                 
----------|---------|----------|---------|---------|-------------------
Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        1.794 s, estimated 3 s
Ran all test suites.
```

This looks like an issue with your local yarn setup and is not ESLint related.
